geoprocessing/src/modules/tile/tile.service.ts

```diff
@@ -0,0 +1,313 @@
+   * @todo add constructor
+   * @todo move generation of specific query for each point to the api. Generate this query with the query builder
+   * @todo fix geometry issue with gid_0 = 'ATA'. Once is fixed, remove this condition from the query.
+   * @description The default base query builder
+   */
+  private readonly logger: Logger = new Logger(TileService.name);
+  defaultGetBaseQuery: GetBaseQuery = ({
+    x,
+    y,
+    z,
+    table,
+    geometry,
+    extent,
+    buffer,
+    customQuery,
+    attributes,
+  }: IBaseQueryInput) => `
+  SELECT
+    ${attributes},
+    ST_AsMVTGeom(
+      -- Geometry from table
+      ST_Transform(ST_RemoveRepeatedPoints(${geometry}, 0.1), 3857),
+      -- MVT tile boundary
+      ST_TileEnvelope(${z}, ${x}, ${y}),
+      -- Extent
+      ${extent},
+      -- Buffer
+      ${buffer},
+      -- Clip geom
+      true
+    ) AS mvt_geom
+  FROM ${table}
+  WHERE
+    ST_Intersects(ST_Transform(ST_TileEnvelope(${z}, ${x}, ${y}), 4326), ${geometry})
+    and ${customQuery}
+    and not gid_0 ='ATA'
+  `;
+
+  /**
+   * @description Creates the query for tile
+   */
+  createQueryForTile({
+    z,
+    x,
+    y,
+    maxZoomLevel,
+    table,
+    geometry,
+    extent,
+    buffer,
+    customQuery,
+    attributes,
+    getBaseQuery = this.defaultGetBaseQuery,
+  }: ITileQueryInput): string {
+    const queryParts: string[] = [];
+    queryParts.push(
+      `WITH tile AS (${getBaseQuery({
+        x,
+        y,
+        z,
+        table,
+        geometry,
+        maxZoomLevel,
+        buffer,
+        extent,
+        customQuery,
+        attributes,
+      })})`,
+    );
+
+    const sql = `${queryParts.join(
+      ',\n',
+    )}\nSELECT ST_AsMVT(tile, 'layer0',  ${extent}, 'mvt_geom') AS mvt FROM tile`;
+
+    return sql;
+  }
+
+  /**
+   * BuilDs tile query.
+   * @param z The zoom level ranging from 0 - 20
+   * @param x The tile x offset on Mercator Projection
+   * @param y The tile y offset on Mercator Projection
+   * @return the resulting string query.
+   */
+  buildQuery(
+    z: number,
+    x: number,
+    y: number,
+    table: string,
+    geometry: string,
+    extent: number,
+    buffer: number,
+    maxZoomLevel: number,
+    customQuery: string,
+    attributes: string,
+  ): string {
+    let query = '';
+
+    z = parseInt(`${z}`, 10);
+    if (isNaN(z)) {
+      throw new Error('Invalid zoom level');
+    }
+
+    x = parseInt(`${x}`, 10);
+    y = parseInt(`${y}`, 10);
+    if (isNaN(x) || isNaN(y)) {
+      throw new Error('Invalid tile coordinates');
+    }
+
+    try {
+      query = this.createQueryForTile({
+        z,
+        x,
+        y,
+        maxZoomLevel,
+        table,
+        geometry,
+        extent,
+        buffer,
+        customQuery,
+        attributes,
+      });
+      this.logger.debug(`Create query for tile: ${query}`);
+    } catch (error) {
+      this.logger.error(`Error getting the query: ${error}`);
+    }
+    return query;
+  }
```

Comment:
Agree, this can be simplified by applying a validator. I've tried to apply it in both here and the admin-areas controller with not success. Maybe we can discuss this further.